src/shogun/classifier/mkl/MKL.cpp

```diff
@@ -262,6 +264,49 @@ CMKL::~CMKL()
 	SG_UNREF(svm);
 }
 
+void CMKL::register_params()
+{
+	// Register parameters
+	SG_ADD((CSGObject**) &svm, "svm", "wrapper svm", MS_NOT_AVAILABLE);
+	SG_ADD(&C_mkl, "C_mkl", "C mkl", MS_NOT_AVAILABLE);
+	SG_ADD(&mkl_norm, "mkl_norm", "norm used in mkl", MS_NOT_AVAILABLE);
+	SG_ADD(&ent_lambda, "ent_lambda", "elastic net sparsity trade-off parameter", MS_NOT_AVAILABLE);
+	SG_ADD(&mkl_block_norm, "mkl_block_norm", "mkl sparse trade-off parameter", MS_NOT_AVAILABLE);
+	m_parameters->add_vector(&beta_local, &beta_local_size, "beta_local", "subkernel weights on L1 term of elastic net mkl");
+	SG_ADD(&mkl_iterations, "mkl_iterations", "number of mkl steps", MS_NOT_AVAILABLE);
+	SG_ADD(&mkl_epsilon, "mkl_epsilon", "mkl epsilon", MS_NOT_AVAILABLE);
+	SG_ADD(&interleaved_optimization, "interleaved_optimization", "whether to use mkl wrapper or interleaved opt.", MS_NOT_AVAILABLE);
+	SG_ADD(&w_gap, "w_gap", "gap between interactions", MS_NOT_AVAILABLE);
+	SG_ADD(&rho, "rho", "objective after mkl iterations", MS_NOT_AVAILABLE);
+	SG_ADD(&lp_initialized, "lp_initialized", "if lp is Initialized", MS_NOT_AVAILABLE);
+
+	// Missing: self (3rd party specific, handled in clone())
+}
+
+CSGObject* CMKL::clone()
+{
+	CMKL* cloned = (CMKL*) CSGObject::clone();
+#ifdef USE_GLPK
+	// GLPK params
+	if (self->lp_glpk_parm != NULL)
+	{
+		cloned->self->lp_glpk_parm = SG_MALLOC(glp_smcp, 1);
+		*cloned->self->lp_glpk_parm = *self->lp_glpk_parm;
+	}
+	// GLPK problem
+	if (self->lp_glpk != NULL)
+	{
+		cloned->self->lp_glpk = glp_create_prob();
+		glp_copy_prob(cloned->self->lp_glpk, self->lp_glpk, GLP_ON);
+	}
+#endif
+#ifdef USE_CPLEX
+	// Shogun currently doesn't compile with CPLEX
```

Comment:
please add `#error` here.... just in case if one day somebody fixes the rest of the cplex code, but forgets this part :)